.github/workflows/test_lint.yml

```diff
@@ -7,32 +7,39 @@ jobs:
     defaults:
       run:
         working-directory: ./app_feup
-    steps: 
+    steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-java@v1
         with:
           java-version: '12.x'
       - uses: subosito/flutter-action@v1
         with:
           flutter-version: '2.0.1'
+
+      - name: Cache pub dependencies
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.FLUTTER_HOME }}/.pub-cache
+          key: ${{ runner.os }}-pub-${{ github.ref }}-${{ hashFiles('**/pubspec.lock') }}
+          restore-keys: ${{ runner.os }}-pub-${{ github.ref }}-
+
       - run: flutter pub get
       - run: flutter analyze --no-pub --preamble .
-      
+
   test:
     name: 'Test'
     runs-on: ubuntu-latest
     defaults:
       run:
         working-directory: ./app_feup
-    steps: 
+    steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-java@v1
         with:
           java-version: '12.x'
```

Comment:
Shouldn't we use the same java version as when deploying to the Google Play Store? Just to make sure no bugs slip through when running the pull request checks...